src/hole.rs

```diff
@@ -40,6 +40,38 @@ impl HoleList {
         }
     }
 
+    /// Places a hole of a given size at a given address after a given hole
+    unsafe fn put_hole(hole: &mut Hole, addr: usize, size: usize) {
+        mem::forget(mem::replace(&mut *(addr as *mut Hole),
```

Comment:
Not sure if it is really needed here, but isn't it the simillar case as in the new call? It seems to work without the forget call though(because Hole does not have a drop Trait?). Still exploring the mysteries of rust :)